superset/db_engine_specs/bigquery.py

```diff
@@ -738,7 +738,7 @@ def _get_fields(cls, cols: list[ResultSetColumnType]) -> list[Any]:
     @classmethod
     def parse_error_exception(cls, exception: Exception) -> Exception:
         try:
-            return Exception(str(exception).splitlines()[0].strip())
+            return type(exception)(str(exception).splitlines()[0].strip())
```

Comment:
This method `parse_error_exception` is very confusing... I think it's really unlikely that `str(exception).splitlines()[0].strip()` will raise an exception — it only happens if `str(exception)` is `''` or if there's some shennanigans in the exception class `__str__` method.

I think what the method is trying to do is limit the exception message to the first line, but there are better ways to do that. But it's OK, we can refactor later.

Reply:
The issue I was running into was when a mapped `SupersetDBAPIError` was being passed into this method, it would then be remapped to a base class `Exception` type with the message preserved. I was trying to preserve the Exception type as well as the message